### PR TITLE
Part 4 Development Branch

### PR DESCRIPTION
**Pull request by transistasis**
In this pull request:
* Completed the requirements for Part 4
* Verified that the outputs match
* Confirmed that there are no errors or warnings

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -460,6 +396,154 @@ IntType& IntType::divide(const int rhs)
...
+// The x and y values stored in the Point class are of type float
+x(static_cast<float>(x_)),
+y(static_cast<float>(y_))
+{}
+
+Point::Point(IntType& x_, IntType& y_) :
+// The x and y values stored in the Point class are of type float
+x(static_cast<float>(x_)),
+y(static_cast<float>(y_))
+{}
```
Comment (matkatmusic) [inline main.cpp:514]: use a `Delegating Constructor`
https://en.cppreference.com/w/cpp/language/constructor
```
Point::Point(const T& a, const T& b) : Point( static_cast<float>(a), static_cast<float>(b) ) { }
```
- [ ] revise

File: main.cpp [hunk 2]
```diff
@@ -460,6 +396,154 @@ IntType& IntType::divide(const int rhs)
...
+{
+    x *= m;
+    y *= m;
+    return *this;
+}
+
+Point& Point::multiply(FloatType& m)
+{
+    // The float() operator defined in FloatType will return a float value
+    return multiply(float(m));
```
Comment (matkatmusic) [inline main.cpp:526]: use `static_cast<>`, not `float()`
`float(m)` is not calling `operator float()` in `FloatType`
- [ ] revise

File: main.cpp [hunk 3]
```diff
@@ -460,6 +396,154 @@ IntType& IntType::divide(const int rhs)
...
+{
+    // Casting doubles to floats will guarantee that equivalent data types are being operated on
+    return multiply(static_cast<float>(m));
+}
+
+Point& Point::multiply(IntType& m)
+{
+    // Casting integers to floats will guarantee that equivalent data types are being operated on
+    return multiply(static_cast<float>(m));
+}
```
Comment (matkatmusic) [inline main.cpp:539]: pass by const reference.  Are you planning on modifying `m`??
- [ ] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]

File: main.cpp (hunk 1 above)
Comment (transistasis) [reply main.cpp:514]: Ah, I missed this. Fixed.

File: main.cpp (hunk 3 above)
Comment (transistasis) [reply main.cpp:539]: I made these `const` arguments since they won't be modified.

**Review by matkatmusic — APPROVED** [review]

**Pull request merged**